### ISSUE TITLE
On the "Data" page, show only the public data collections

### ISSUE THREAD

**Issue by nutjob4life** [EDRN/P5#298]
Collections like LTP2 should not be displayed.

**Comment by nutjob4life** [comment]
@hoodriverheather as you figured out [for this presentation](https://docs.google.com/presentation/d/1UW7Tdy1kUJRxxVgJyZLx6Kqfw3nc3sIk5XFpgy789FI/edit#slide=id.g2984fb1f036_0_5)—and as I totally forgot, doh!—the P5 portal already filters the collections [on the data page](https://edrn.nci.nih.gov/data-and-resources/data/) based on who's logged in and what collections they would be allowed to see.

So there's nothing to change on the Data page; but that does open up two more issues.


### Investigator Pages

Do we want to limit the numbers of data collections appearing on investigator ("vanity") pages to what the logged-in user would be able to see?

My feeling is **no** for two reasons:

1. These numbers are already vanishingly small. Many investigators have a pittance of numbers, just a few `1`s scattered about. Ziding is an outlier in that he's got one protocol with _three_. If we limited those to what the logged-in user could see, those numbers would drop even further.
2. The numbers don't reveal the names of the data collections, so there's no risk.

What do you think?


### Protocol Data Collections

Take a look at [Marc Lenburg's page](https://edrn-dev.jpl.nasa.gov/portal/renaissance/about-edrn/sites/594-boston-university/lenburg-marc/)'s on the dev portal. It shows for the "Validation of Molecular Biomarkers for the Early Detection of Lung Cancer in the setting of Indeterminate Pulmonary Nodules (Lung Team Project №2)" protocol that he has one data collection (which by my reasoning above, should be okay to display).

However, if you then click the link for that collection—as a _non-logged in user_—you're taken to that protocol page which proudly announces that its one data collections is the private LTP2 collection.

My feeling here is **yes**, we should limit what collections appear here, since a _name_ of a collection is visibile to everyone. We might even show something like the following made-up example:

> ### Validation of Very Important Markers
>
> Abbreviated Name: VVIM
>
> Lead Investigator: Schmoe, Josephine — Institute for Instituting
>
> #### Abstract
>
> Blah blah blah …
>
> …
>
> #### Data Collections
> 
> - Analysis of pancreatic cancer biomarkers in PLCO set
> - Basophile
> - University of Washington Microarray Data
> 
> There are 2 additional collections of data for this protocol which have not yet been made public.

What do you think?

**Comment by nutjob4life** [comment]
@hoodriverheather based on [my comment above ↑](https://github.com/EDRN/P5/issues/298#issuecomment-1805882129) I went ahead and implemented this in the [dev portal](https://edrn-dev.jpl.nasa.gov/portal/renaissance/).

- The Data page remains the same; if you're logged in, you might see additional collections depending on access
- Investigator pages show the counts of collections, public or otherwise.
- Protocol pages now show public collections and a count of any non-public collections; log in and depending if you've got access, you might see more

Try it out as an anonymous user and as a logged in user to get a feel for it. I like it! What do you think?

**Comment by hoodriverheather** [comment]
@nutjob4life I think this makes sense. I like that the protocol page shows the comment that there's a private data collection. Do you think we should include any additional wording that tells the user to login? It is tricky, because it depends on if you have access.

**Comment by nutjob4life** [comment]
@hoodriverheather I've updated the [dev portal](https://edrn-dev.jpl.nasa.gov/portal/renaissance/) again and changed the message at the bottom of a protocol page as follows:

- If there's one hidden collection and you're not logged in: `There is one collection of data that has not yet been made public. Logging in may you give you access to the names of the non-public data collections.` — "Logging in" is a hyperlink.
- If there's one hidden collection and you are logged in: `There is one collection of data that has not yet been made public. You are currently logged in, but not with an account that can view the names of the non-public data collections. Logging in with a different account may give you access.` — "Logging in with a different account" is a hyperlink.
- If there's more than one hidden collection and you're not logged in: `There are N collections of data that have not yet been made public. Logging in may you give you access to the names of the non-public data collections.` — "Logging in" is a hyperlink and `N` is a number.
- If there's more than one hidden collection and you're logged in: `There are N collections of data that have not yet been made public. You are currently logged in, but not with an account that can view the names of the non-public data collections. Logging in with a different account may give you access.` — "Logging in with a different account" is a hyperlink. "Logging in with a different account" is a hyperlink and `N` is a number.

Whew. 

Anyway, I used the specific wording of "names of data collections" to emphasize that _even if you do log into the portal_, all you get is access to the _names of the collections_. Because if you click on a link to get the details, you leave the portal and go to LabCAS, where you'll have to log in again.

Sound good?
